### PR TITLE
EXIF IDF pointer

### PR DESCRIPTION
**Pull request by IldarKhayrutdinov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes for #1921

### CODE REVIEW

**Comment by antonfirsov** [comment]
Can we add a test case based on the #1921 repro steps?

**Comment by IldarKhayrutdinov** [comment]
I needs to additional reviewing of EXIF writer methods, it seems there are problems of IFD writing (regarding next IFD pointers and gps sub IFD). Next IFD pointer should be written after each IFD, but at the moment it's written only after the last IFD.

File: src/ImageSharp/Metadata/Profiles/Exif/ExifWriter.cs [hunk 1]
```diff
@@ -56,39 +55,38 @@ public byte[] GetData()
...
+            byte[] result = new byte[length];
 
             int i = 0;
 
             // The byte order marker for little-endian, followed by the number 42 and a 0
             ExifConstants.LittleEndianByteOrderMarker.CopyTo(result.AsSpan(start: i));
             i += ExifConstants.LittleEndianByteOrderMarker.Length;
 
-            uint ifdOffset = ((uint)i - startIndex) + 4U;
-            uint thumbnailOffset = ifdOffset + ifdLength + exifLength + gpsLength;
```
Comment (IldarKhayrutdinov) [inline src/ImageSharp/Metadata/Profiles/Exif/ExifWriter.cs:84]: in fact, the thumbnail isn't written, this is an unnecessary line

**Review by JimBobSquarePants — APPROVED** [review]
This looks great! Thanks for stepping up 👍

**Pull request merged**